tests/components/webostv/test_media_player.py

```diff
@@ -1,50 +1,211 @@
+        (SERVICE_VOLUME_DOWN, "volume_down"),
+        (SERVICE_MEDIA_PLAY, "play"),
+        (SERVICE_MEDIA_PAUSE, "pause"),
+        (SERVICE_MEDIA_STOP, "stop"),
+    ],
+)
+async def test_services(hass, client, service, client_call):
+    """Test simple services without parameters."""
+    await setup_webostv(hass)
 
-    client.set_mute.assert_called_once()
+    data = {ATTR_ENTITY_ID: ENTITY_ID}
+    assert await hass.services.async_call(MP_DOMAIN, service, data, True)
+
+    getattr(client, client_call).assert_called_once()
+
+
+async def test_media_play_pause(hass, client):
+    """Test media play pause service."""
+    await setup_webostv(hass)
 
+    data = {ATTR_ENTITY_ID: ENTITY_ID}
 
-async def test_select_source_with_empty_source_list(hass, client):
+    # After init state is playing - check pause call
+    assert await hass.services.async_call(
+        MP_DOMAIN, SERVICE_MEDIA_PLAY_PAUSE, data, True
+    )
+
+    client.pause.assert_called_once()
+    client.play.assert_not_called()
+
+    # After pause state is paused - check play call
+    assert await hass.services.async_call(
+        MP_DOMAIN, SERVICE_MEDIA_PLAY_PAUSE, data, True
+    )
+
+    client.play.assert_called_once()
+    client.pause.assert_called_once()
+
+
+@pytest.mark.parametrize(
+    "service, client_call",
+    [
+        (SERVICE_MEDIA_NEXT_TRACK, ("fast_forward", "channel_up")),
+        (SERVICE_MEDIA_PREVIOUS_TRACK, ("rewind", "channel_down")),
+    ],
+)
+async def test_media_next_previous_track(
+    hass, client, service, client_call, monkeypatch
+):
+    """Test media next/previous track services."""
+    await setup_webostv(hass)
+
+    # check channel up/down for live TV channels
+    data = {ATTR_ENTITY_ID: ENTITY_ID}
+    assert await hass.services.async_call(MP_DOMAIN, service, data, True)
+
+    getattr(client, client_call[0]).assert_not_called()
+    getattr(client, client_call[1]).assert_called_once()
+
+    # check next/previous for not Live TV channels
+    monkeypatch.setattr(client, "current_app_id", "in1")
+    data = {ATTR_ENTITY_ID: ENTITY_ID}
+    assert await hass.services.async_call(MP_DOMAIN, service, data, True)
+
+    getattr(client, client_call[0]).assert_called_once()
+    getattr(client, client_call[1]).assert_called_once()
+
+
+async def test_select_source_with_empty_source_list(hass, client, caplog):
     """Ensure we don't call client methods when we don't have sources."""
     await setup_webostv(hass)
+    await client.mock_state_update()
 
     data = {
         ATTR_ENTITY_ID: ENTITY_ID,
         ATTR_INPUT_SOURCE: "nonexistent",
     }
-    await hass.services.async_call(MP_DOMAIN, SERVICE_SELECT_SOURCE, data)
-    await hass.async_block_till_done()
+    assert await hass.services.async_call(MP_DOMAIN, SERVICE_SELECT_SOURCE, data, True)
 
     client.launch_app.assert_not_called()
     client.set_input.assert_not_called()
+    assert f"Source nonexistent not found for {TV_NAME}" in caplog.text
```

Comment:
Usually when I do string matching, I check just enough of the message to be sure the right one happened since it makes a bit easier to maintain.